### DISCUSSION TITLE
How do I get a access token outside of useLogto() with @logto/react

### DISCUSSION THREAD

**Discussion by isaced** [logto-io/logto#6804] [Q&A]
I'm trying to add an auth header to my fetch function but I'm finding it difficult.

I could use `useLogto()` but this is awkward I want to do something like

```js
import { getAccessToken } from "@logto/react";

fetch('whatever', { headers: { Authorization: await getAccessToken() } })
```

Or is there any way I can get the `LogtoClient` instance in React to call its `getAccessToken()` method in my network library?

See also:
- https://github.com/orgs/clerk/discussions/1751
- https://github.com/clerk/javascript/pull/3420

**Comment by isaced** [comment]
I've found a rather **tricky** approach, customize the `LogtoClientClass` through the `LogtoProvider` to globally hold its instance object, I'm not sure if there are any other more elegant ways.

```ts
// Path: src/main.tsx
import { LogtoProvider, LogtoConfig } from "@logto/react";
import LogtoClient from "@logto/browser";

// Store the global LogtoClient instance
let globalLogtoClientInstance: LogtoClient | null = null;

// Custom LogtoClient class to store the instance
class CustomLogtoClient extends LogtoClient {
  constructor(config: LogtoConfig) {
    super(config);
    globalLogtoClientInstance = this;
  }
}

ReactDOM.createRoot(document.getElementById("root")!).render(
  <React.StrictMode>
      <LogtoProvider config={config} LogtoClientClass={CustomLogtoClient}>
        <RouterProvider router={router} />
      </LogtoProvider>
  </React.StrictMode>
);
```

```ts
// Path: src/services/network.ts
class NetworkService {
  async fetchSomething() {
    // Get the access token from the global LogtoClient instance
    const accessToken = globalLogtoClientInstance?.getAccessToken();
    // ...
  }
}
```

**Comment by charIeszhao** [comment]
Your approach is a bit tricky, but it works.

Can you tell me why do you want to do this in the first place? The `useLogto` is a React hook, and can be used in either a React component or another React hook. If you want to fetch access token in another non-hook file (e.g. /services/network.ts), you may consider refactoring your utils to a React hooks instead.
